lib/Parse/ASTGen.cpp

```diff
@@ -10,80 +10,738 @@
+      SourceRange Range{LAngleLoc, RAngleLoc};
+      auto ArgsAST = generate(Args, Loc);
+      return GenericIdentTypeRepr::create(Context, IdentifierLoc, Identifier,
+                                          ArgsAST, Range);
+    }
+  }
+  return new (Context) SimpleIdentTypeRepr(IdentifierLoc, Identifier);
+}
+
+TypeRepr *ASTGen::generate(SimpleTypeIdentifierSyntax Type, SourceLoc &Loc) {
+  if (Type.getName().getTokenKind() == tok::kw_Any) {
+    auto AnyLoc = advanceLocBegin(Loc, Type.getName());
+    return CompositionTypeRepr::createEmptyComposition(Context, AnyLoc);
+  }
+
+  return generateSimpleOrMemberIdentifier(Type, Loc);
+}
+
+TypeRepr *ASTGen::generate(MemberTypeIdentifierSyntax Type, SourceLoc &Loc) {
+  return generateSimpleOrMemberIdentifier(Type, Loc);
+}
+
+TypeRepr *ASTGen::generate(DictionaryTypeSyntax Type, SourceLoc &Loc) {
+  TypeRepr *ValueType = generate(Type.getValueType(), Loc);
+  TypeRepr *KeyType = generate(Type.getKeyType(), Loc);
+  auto LBraceLoc = advanceLocBegin(Loc, Type.getLeftSquareBracket());
+  auto ColonLoc = advanceLocBegin(Loc, Type.getColon());
+  auto RBraceLoc = advanceLocBegin(Loc, Type.getRightSquareBracket());
+  SourceRange Range{LBraceLoc, RBraceLoc};
+  return new (Context) DictionaryTypeRepr(KeyType, ValueType, ColonLoc, Range);
+}
+
+TypeRepr *ASTGen::generate(ArrayTypeSyntax Type, SourceLoc &Loc) {
+  TypeRepr *ElementType = generate(Type.getElementType(), Loc);
+  auto LBraceLoc = advanceLocBegin(Loc, Type.getLeftSquareBracket());
+  auto RBraceLoc = advanceLocBegin(Loc, Type.getRightSquareBracket());
+  SourceRange Range{LBraceLoc, RBraceLoc};
+  return new (Context) ArrayTypeRepr(ElementType, Range);
+}
+
+TypeRepr *ASTGen::generate(MetatypeTypeSyntax Type, SourceLoc &Loc) {
+  TypeRepr *BaseType = generate(Type.getBaseType(), Loc);
+  auto TypeOrProtocol = Type.getTypeOrProtocol();
+  auto TypeOrProtocolLoc = advanceLocBegin(Loc, TypeOrProtocol);
+  if (TypeOrProtocol.getText() == "Type")
+    return new (Context) MetatypeTypeRepr(BaseType, TypeOrProtocolLoc);
+  return new (Context) ProtocolTypeRepr(BaseType, TypeOrProtocolLoc);
+}
+
+TypeRepr *ASTGen::generate(OptionalTypeSyntax Type, SourceLoc &Loc) {
+  TypeRepr *WrappedType = generate(Type.getWrappedType(), Loc);
+  auto QuestionLoc = advanceLocBegin(Loc, Type.getQuestionMark());
+  return new (Context) OptionalTypeRepr(WrappedType, QuestionLoc);
+}
+
+TypeRepr *ASTGen::generate(ImplicitlyUnwrappedOptionalTypeSyntax Type,
+                           SourceLoc &Loc) {
+  TypeRepr *WrappedType = generate(Type.getWrappedType(), Loc);
+  auto ExclamationLoc = advanceLocBegin(Loc, Type.getExclamationMark());
+  return new (Context)
+      ImplicitlyUnwrappedOptionalTypeRepr(WrappedType, ExclamationLoc);
+}
+
+TypeRepr *ASTGen::generate(UnknownTypeSyntax Type, SourceLoc &Loc) {
+  auto ChildrenCount = Type.getNumChildren();
+
+  if (ChildrenCount == 3 || ChildrenCount == 4) {
```

Comment:
Could you add comments for each `Unknown*Syntax` conversion branch?
For example this branch could be:
```
// Constract 'ArrayTypeRepr' from obsoleted syntax. i.e.
//     type-simple '[' ...
```

Reply:
Ah sorry, these comments are to https://github.com/apple/swift/pull/26478

Reply:
No worries, I've implemented and pushed both of your suggestions to #26478.